Automattic/wp-calypso
# Jetpack connect redirect wp admin

Right now, the new jetpack authorization screen is offering the chance of checking Jetpack plans to any user that connects a 3rd party site. The old authorization screen didn't use to do it, it just redirected users back to their site once the connection process was complete.

More sooner than later we are going to have a new post-connection landing page to substitute this redirection to plans page, but meanwhile, this PR deactivates the plans page's link and redirect users directly back to their wp-admin once the connection is complete. This happens only for users that have started their connection flow from wp-admin, not for those using jetpack-connect (who will still see the link to the plans page).

![untitled screencast 11](https://cloud.githubusercontent.com/assets/1554855/15013208/5fcfeed0-11fe-11e6-8594-e60266632764.gif)

How to test
========
1. This is a tricky one. You need to be running a .com API sandbox and modify `wp-content/mu-plugins/jetpack/class.jetpack-server-version.php`'s get_calypso_url method (line 671) to make it point to your calypso.localhost in every case. 
2. You need to be using a jetpack that is connected to your .com sandbox
3. Disconnect your jetpack and start a connection from wp-admin. After approving the connection, you should be redirected back to your wp-admin
4. This WILL NOT WORK if you have used jetpack-connect with that site in the last hour 

ping @roccotripaldi @ryelle @oskosk
